### PR TITLE
Fix #137, fitToMarkers

### LINKED ISSUE #137: fit-to-markers and fitToMarkers are broken
In the case of fit-to-markers (which is what shows up in the example usage), it will not let you zoom or move the map.

In the case of fitToMarkers (which is what shows up in the documentation), it doesn't actually do anything at all from what I can tell.

### PR DESCRIPTION
**Pull request by atotic**
2 fixes:
- on resize(), if fitToMarkers redo _fitToMarkersChanged(). Without this, you see nothing
- do not _updateMarkers if marker list has not changed. google maps create tons
  of update events, and it feels very expensive, and prevents scrolling of fit-to-markers

### CODE REVIEW

File: google-map.html [hunk 1]
```diff
@@ -378,9 +378,23 @@
     },
 
     _updateMarkers: function() {
-      this.markers = Array.prototype.slice.call(
+      var newMarkers = Array.prototype.slice.call(
           Polymer.dom(this.$.markers).getDistributedNodes());
 
+      // do not recompute if markers have not been added or removed
+      if (newMarkers.length === this.markers.length) {
+        var added = newMarkers.filter( function(m) {
```
Comment (ebidel) [inline google-map.html]: spacing: `filter(function(m) {`

File: google-map.html [hunk 2]
```diff
@@ -378,9 +378,23 @@
...
+            return this.markers && this.markers.indexOf(m) === -1;
```
Comment (ebidel) [inline google-map.html]: what about removals?

File: google-map.html [hunk 3]
```diff
@@ -378,9 +378,23 @@
...
+          }.bind(this));
+        if (added.length === 0) {
+          return;
+        }
+      }
+
+      this.markers = newMarkers;
+      // this.markers = Array.prototype.slice.call(
```
Comment (ebidel) [inline google-map.html]: remove comment

File: google-map.html (hunk 2 above)
Comment (atotic) [inline google-map.html]: If removed, length would not be the same.

File: google-map.html (hunk 1 above)
Comment (atotic) [inline google-map.html]: dang, I hate the narrow spacing.

**Comment by atotic** [comment]
All fixes in. PTAL

File: google-map.html (hunk 2 above)
Comment (ebidel) [inline google-map.html]: That's true for additions as well. Im confused on the filtering. If the length changed, you always want to recompute:

    if (newMarkers.length === this.markers.length) {
      return;
    }

Comment (atotic) [inline google-map.html]: The goal is not to redo work if markers have not changed. Markers have not changed if:
- if lists are same length, and
- all new markers are found in the old list
Lists could be same length, but markers in the list could be different.

**Comment by ebidel** [comment]
LGTM

**Pull request merged**